src/services.py

```diff
@@ -516,131 +517,254 @@ def ban_user(self, my_user_id, other_user_id):
             )
 
 
-def serialize_chat(chat_room_result, chat_result, buyer, seller, user_id):
-    (author, dealer) = (
-        (seller, buyer) if seller.get("id") == user_id else (buyer, seller)
-    )
-    return {
-        "dealer_id": dealer.get("id"),
-        "created_at": datetime.timestamp(chat_result.get("created_at")),
-        "updated_at": datetime.timestamp(chat_result.get("updated_at")),
-        "author_name": author.get("full_name"),
-        "author_id": author.get("id"),
-        "message": chat_result.get("message"),
-        "chatRoom_id": chat_room_result.get("id"),
-    }
-
-
-class ChatService:
+class OfferService:
     def __init__(self, config):
         self.config = config
 
-    def add_message(self, chat_room_id, message, author_id):
+    def _serialize_offer(self, offer, user_id):
+        return {
+            "id": offer.get("id"),
+            "price": offer.get("price"),
+            "number_of_shares": offer.get("number_of_shares"),
+            "is_buyer_agreeable": offer.get("is_buyer_agreeable"),
+            "is_seller_agreeable": offer.get("is_seller_agreeable"),
+            "is_rejected": offer.get("is_rejected"),
+            "updated_at": datetime.timestamp(offer.get("updated_at")) * 1000,
+            "is_author": True if offer.get("author_id") == user_id else False,
+            "type": "offer",
+        }
+
+    def _serialize_chat_offer(self, chat_room_id, offer, user_id):
+        return {
+            "chat_room_id": chat_room_id,
+            "updated_at": datetime.timestamp(offer.get("updated_at")) * 1000,
+            "new_chat": self._serialize_offer(offer=offer, user_id=user_id),
+        }
+
+    def _get_current_offer(self, session, offer):
+        session.add(offer)
+        session.flush()
+        session.refresh(offer)
+        return offer.asdict()
+
+    def _update_chatroom_datetime(self, session, chat_room, offer):
+        chat_room.updated_at = offer.get("updated_at")
+        session.commit()
+
+    def _verify_user(self, chat_room, user_id, user_type):
+        if user_type == "buyer" and chat_room.buyer_id != user_id:
+            raise ResourceNotOwnedException("Wrong user")
+        if user_type == "seller" and chat_room.seller_id != user_id:
+            raise ResourceNotOwnedException("Wrong user")
```

Comment:
Actually in these helper functions `self` is not used. Let us remove `self` and convert them to static methods.
```python
@staticmethod
def _get_current_offer(session, offer):
  ...
```

Reply:
Same with the `ChatService` below